modules/gateway/gateway.go

```diff
@@ -209,6 +189,15 @@ func (g *Gateway) Address() modules.NetAddress {
 	return g.myAddr
 }
 
+// sessionHeader returns the header for the session of this gateway
+func (g *Gateway) sessionHeader(wantConn bool) sessionHeader {
+	return sessionHeader{
+		GenesisID: types.GenesisID,
+		UniqueID:  g.id,
+		WantConn:  wantConn,
+	}
+}
```

Comment:
hmm. I'm not sure about this. I feel like functions like `connectSessionHandshake` shouldn't have to take a whole `sessionHeader`, since they already know two out of three things (the genesis ID and whether they want a connection). So perhaps they should just take a `gatewayID` instead of a full `sessionHeader`. The only downside is that during testing, you can't change the genesis ID that they'll send/expect. But I don't think that's a huge deal.